lib/chewy/type/adapter/active_record.rb

```diff
@@ -69,13 +69,25 @@ def object_class
       end
 
       ActiveSupport.on_load(:active_record) do
-        if ::ActiveRecord::VERSION::MAJOR >= 5
-          module Rails5
+        if ::ActiveRecord::VERSION::MAJOR < 4
+          module Rails3
             def pluck_ids(scope)
-              scope.except(:includes).distinct.pluck(target.primary_key.to_sym)
+              # for activerecord < 4, pluck can't handle scopes with conditions using includes
+              scope.except(:includes).uniq.pluck(target.primary_key.to_sym)
             end
           end
-          Chewy::Type::Adapter::ActiveRecord.prepend(Rails5)
+          Chewy::Type::Adapter::ActiveRecord.prepend(Rails3)
+        elsif ::ActiveRecord::VERSION::MAJOR >= 4 && ::ActiveRecord::VERSION::MAJOR < 5
+          module Rails4
+            def pluck_ids(scope)
+              if scope.is_a?(::ActiveRecord::Associations::CollectionProxy)
```

Comment:
The `::ActiveRecord::Associations::CollectionProxy` is given (only) from the specs inside `spec/chewy/type/observe_spec.rb`, without knowing why in detail. I only noticed this because of (some) failing specs with the ActiveRecord 4 gemfiles.